compiler/rustc_hir_typeck/src/method/suggest.rs

```diff
@@ -1086,7 +1094,34 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                             *region,
                             ty::TypeAndMut { ty: *t_type, mutbl: mutability.invert() },
```

Comment:
nit: This is will always be `Mutability::Mut`, right? Since it `needs_mut`, so it's not mut already.

Reply:
I believe that there was one case where that wasn't the case that I encountered while testing, but I might be misremembering.

Reply:
I guess it's fine to leave it as-is if you aren't sure